File: backend/models/mentorship.py
from pydantic import BaseModel, Field
from typing import Optional, List
from datetime import datetime
from enum import Enum

class MentorshipStatus(str, Enum):
    PENDING = "pending"
    ACTIVE = "active"
    COMPLETED = "completed"
    CANCELLED = "cancelled"

class SessionStatus(str, Enum):
    SCHEDULED = "scheduled"
    IN_PROGRESS = "in_progress"
    COMPLETED = "completed"
    CANCELLED = "cancelled"
    NO_SHOW = "no_show"

class MentorProfile(BaseModel):
    mentor_id: str
    user_id: str
    specialties: List[str]
    years_experience: int
    hourly_rate: Optional[float] = None
    availability_hours: List[str] = []
    max_mentees: int = 10
    current_mentees: int = 0
    rating: float = 0.0
    total_sessions: int = 0
    response_time_hours: float = 24.0
    is_accepting_mentees: bool = True
    bio: Optional[str] = None
    linkedin_url: Optional[str] = None
    created_at: datetime = Field(default_factory=datetime.now)

class MentorshipConnection(BaseModel):
    connection_id: str
    mentor_id: str
    mentee_id: str
    status: MentorshipStatus
    requested_at: datetime = Field(default_factory=datetime.now)
    accepted_at: Optional[datetime] = None
    completed_at: Optional[datetime] = None
    goals: List[str] = []
    notes: Optional[str] = None

class MentorshipSession(BaseModel):
    session_id: str
    connection_id: str
    mentor_id: str
    mentee_id: str
    scheduled_at: datetime
    duration_minutes: int = 60
    status: SessionStatus = SessionStatus.SCHEDULED
    meeting_url: Optional[str] = None
    agenda: Optional[str] = None
    notes: Optional[str] = None
    mentor_feedback: Optional[str] = None
    mentee_feedback: Optional[str] = None
    rating: Optional[int] = None
    created_at: datetime = Field(default_factory=datetime.now)

class MentorshipRequest(BaseModel):
    mentor_id: str
    goals: List[str]
    message: Optional[str] = None

class SessionBooking(BaseModel):
    connection_id: str
    scheduled_at: datetime
    duration_minutes: int = 60
    agenda: Optional[str] = None

class MentorResponse(BaseModel):
    mentor_id: str
    user_id: str
    full_name: str
    avatar_url: Optional[str] = None
    specialties: List[str]
    years_experience: int
    rating: float
    total_sessions: int
    hourly_rate: Optional[float] = None
    bio: Optional[str] = None
    is_accepting_mentees: bool

File: frontend/src/hooks/types/mentorship.ts

export interface MentorProfile {
  id: string;
  userId: string;
  expertise: string[];
  yearsOfExperience: number;
  bio?: string;
  maxMentees: number;
  isAcceptingMentees: boolean;
  availability?: string;
}

export interface MentorshipRequest {
  id?: string;
  requester_id: string;
  mentor_id: string;
  message: string;
  status: string;
  created_at?: string;
  focus_areas?: string[];
  industry: string;
  expected_duration?: string;
  goals?: string[];
}

export interface MentorshipRelationship {
  id: string;
  mentor_id: string;
  mentee_id: string;
  status: string;
  focus_areas?: string[];
  goals?: string;
  meeting_frequency?: string;
  feedback?: string;
  created_at: string;
  updated_at: string;
  end_date?: string;
}

export interface MentorshipSession {
  id: string;
  relationship_id: string;
  scheduled_for: string;
  duration_minutes?: number;
  status: string;
  notes?: string;
  mentor_feedback?: string;
  mentee_feedback?: string;
  mentor_rating?: number;
  mentee_rating?: number;
}

export interface MentorshipProgress {
  id: string;
  mentorship_id: string;
  milestone: string;
  notes?: string;
  completed: boolean;
  date: string;
}

export interface MentorshipResource {
  id: string;
  mentorship_id: string;
  title: string;
  type: string;
  url?: string;
  description?: string;
  added_by: string;
  added_at: string;
  completed?: boolean;
}
